solutions/fully-configurable/main.tf

```diff
@@ -48,7 +42,7 @@ resource "ibm_iam_authorization_policy" "kms_policy" {
   target_service_name         = local.kms_service_name
   target_resource_instance_id = local.existing_kms_guid
   roles                       = ["Reader"]
-  description                 = "Allow all Secrets Manager instances in the resource group ${module.resource_group[0].resource_group_id} in the account ${data.ibm_iam_account_settings.iam_account_settings[0].account_id} to read from the ${local.kms_service_name} instance GUID ${local.existing_kms_guid}"
+  description                 = "Allow all Secrets Manager instances in the resource group ${module.resource_group.resource_group_id} in the account ${data.ibm_iam_account_settings.iam_account_settings[0].account_id} to read from the ${local.kms_service_name} instance GUID ${local.existing_kms_guid}"
```

Comment:
I think we should add the support to scope the KMS IAM auth policy to the exact key in the module before we merge this PR. That way we won't have to worry about a further update to the auth policy in the DA later. Seems there is a [PR](https://github.com/terraform-ibm-modules/terraform-ibm-secrets-manager/pull/295) open for a few weeks. Can we get an update on this?

Once that is in, we should also scope to the key here in this DA. Note that an extra viewer policy is required when using HPCS (you can see it in above PR) and scoping to exact key.